plugins/usi-dock/fu-usi-dock-mcu-device.c

```diff
@@ -568,18 +599,19 @@ fu_usi_dock_mcu_device_write_firmware_with_idx(FuUsiDockMcuDevice *self,
 					 TAG_TAG2_CMD_SPI,
 					 &cmd,
 					 sizeof(cmd),
-					 &val,
-					 sizeof(val),
+					 NULL,
+					 0x0,
 					 error))
```

Comment:
nah, i mean this whole thing -- it's always called at least once with retry.

Reply:
Do you mean to omit this command? I had to make sure that the order of these commands would behave the same as on other platforms, and we would only write to the external flash once. We need this retry mechanism.

Reply:
i mean you could omit the:

	/* initial external flash */
	cmd = USBUID_ISP_DEVICE_CMD_FWBUFER_INITIAL;
	if (!fu_usi_dock_mcu_device_txrx(self,
					 TAG_TAG2_CMD_SPI,
					 &cmd,
					 sizeof(cmd),
					 &val,
					 sizeof(val),
					 error))
		return FALSE;
	if (val != SPI_STATE_READY) {
		g_set_error(error,
			    G_IO_ERROR,
			    G_IO_ERROR_INVALID_DATA,
			    "invalid state for CMD_FWBUFER_INITIAL, got 0x%02x",
			    val);
		return FALSE;
	}

and just rely on it being called in the fu_usi_dock_mcu_device_wait_for_spi_initial_ready_cb() loop.